lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -158,16 +144,51 @@ protected function doGetStats()
         );
     }
 
+    private function setDirectory($directory)
+    {
+        if ( ! $this->createPathIfNeeded($directory, 0777)) {
+            throw new \InvalidArgumentException(sprintf(
+                'The directory "%s" does not exist and could not be created.',
+                $directory
+            ));
+        }
+
+        if (function_exists('posix_geteuid')) {
+            $userData = posix_getpwuid(posix_geteuid());
+            $directory .= DIRECTORY_SEPARATOR
+                . $userData['name'];
+
+            if ( ! $this->createPathIfNeeded($directory)
+                || fileowner($directory) !== posix_geteuid()
```

Comment:
It would. It's mooted by [the core team's blog post on umasks](http://www.doctrine-project.org/2015/08/31/security_misconfiguration_vulnerability_in_various_doctrine_projects.html), so I'll go ahead and close it out.